core/pom.xml

```diff
@@ -80,11 +81,35 @@
                           org.osgi.framework,
                           org.osgi.util.tracker,
                         ]]></Import-Package>
+                        <!--
+                        This is exports `org.slf4j` APIs so that this is the only instance of slf4j
+                        initialized. Otherwise, each bundle will re-initialize their own instance
+                        in each bundle. This is fine when using stateless appenders, but this causes
+                        unpredictable results when using stateful appenders such as
+                        `ch.qos.logback.core.rolling.RollingFileAppender`.
+
+                        Ideally, this would not export `ch.qos.logback` APIs; however, the
+                        persistence backends include dependencies (`cassandra-all` and `dse-db`)
+                        that use logback APIs directly e.g. calling
+                        `org.slf4j.LoggerFactory.getLogger(...)` and casting it to a logback
+                        `ch.qos.logback.classic.Logger`. This results in casting errors without
+                        exporting these APIs. Only the persistence backends import these.
+
+                        If it can be helped, no other bundle should include any or use logback
+                        dependencies. Instead they should stick to using slf4j APIs.
+                        -->
                         <Export-Package><![CDATA[
                           io.stargate.core.*,
                           com.codahale.metrics,
                           com.codahale.metrics.*,
-                          io.micrometer.core.*
+                          io.micrometer.core.*,
+                          org.slf4j,
+                          org.slf4j.event,
+                          org.slf4j.helpers,
+                          org.slf4j.spi,
+                          ch.qos.logback.classic,
+                          ch.qos.logback.classic.helpers,
+                          ch.qos.logback.classic.spi,
```

Comment:
Is there a need to export this at all if nobody is directly using and depending on the `logback`?

EDIT: OK I see now persistence is importing.. What is the reason? Is persistence setting up something in logback?

Reply:
I address  this in the comment above.

Reply:
Example of `cassandra-all` using logback: https://github.com/apache/cassandra/blob/cassandra-3.11/src/java/org/apache/cassandra/utils/logging/LogbackLoggingSupport.java#L69-L70

Reply:
OK was not aware of the `cassandra-all` using it..